torch/_dynamo/output_graph.py

```diff
@@ -354,6 +362,10 @@ def guards(self) -> Set[Guard]:
     def nn_modules(self) -> Dict[str, torch.nn.Module]:
         return self.tracing_context.module_context.nn_modules
 
+    def save_global_state(self):
+        global_state = self.tracing_context.global_context.global_state
+        global_state["is_grad_enabled"] = torch.is_grad_enabled()
```

Comment:
basic question: there's other global state, like e.g. current autocast context. Should this function be checkpointing that too?

Reply:
Yes, thats what I was thinking too. Yes, we should. Thats why I created this central place to handle global states. We will add more cases here as they come.